src/ipc/ipc4/helper.c

```diff
@@ -366,6 +369,8 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 		goto err;
 	}
 
+	irq_local_enable(flags);
```

Comment:
You need to enable interrupts back in case of `goto err;`

Reply:
Thanks @pblaszko fixed now. In fact I also added another patch to free the allocated buffer in case of all errors.

Reply:
Thanks! Now looks good to me.